waspc/src/Wasp/Generator/FileDraft/CopyDirFileDraft.hs

```diff
@@ -0,0 +1,34 @@
+module Wasp.Generator.FileDraft.CopyDirFileDraft
+  ( CopyDirFileDraft (..),
+  )
+where
+
+import Control.Monad (when)
+import StrongPath
+  ( Abs,
+    Dir',
+    Path',
+    Rel,
+    (</>),
+  )
+import qualified StrongPath as SP
+import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.FileDraft.Writeable
+import Wasp.Generator.FileDraft.WriteableMonad (WriteableMonad (copyDirectoryRecursive), doesDirectoryExist)
+
+-- | File draft based on another dir that is to be recursively copied.
+data CopyDirFileDraft = CopyDirFileDraft
+  { -- | Path where the dir will be copied to.
+    _dstPath :: !(Path' (Rel ProjectRootDir) Dir'),
+    -- | Absolute path of source dir to copy.
+    _srcPath :: !(Path' Abs Dir')
+  }
+  deriving (Show, Eq)
+
+instance Writeable CopyDirFileDraft where
+  write absDstDirPath draft = do
```

Comment:
While it never seemed to matter in testing, just for safety and to mirror the other `*FileDraft` types I may need to add:
```haskell
createDirectoryIfMissing True (SP.fromAbsDir $ SP.parent dstPathAbsDir)
```

Reply:
Never came up in testing since the `db` dir always existed. But need it in there for other use cases/safety.

Reply:
Added